docs/source/error_code_list.rst

```diff
@@ -0,0 +1,502 @@
+is compatible with the list type that is inferred from the surrounding
+context.
+
+Example:
+
+.. code-block:: python
+
+    from typing import List
+
+    # Error: List item 0 has incompatible type "int"; expected "str"  [list-item]
+    a: List[str] = [0]
+
+Check dict items [dict-item]
+----------------------------
+
+When constructing a dictionary using ``{key: value, ...}`` or ``dict(key=value, ...)``,
+mypy checks that each key and value is compatible with the dictionary type that is
+inferred from the surrounding context.
+
+Example:
+
+.. code-block:: python
+
+    from typing import Dict
+
+    # Error: Dict entry 0 has incompatible type "str": "str"; expected "str": "int"  [dict-item]
+    d: Dict[str, int] = {'key': 'value'}
+
+Check TypedDict items [typeddict-item]
+--------------------------------------
+
+When constructing a TypedDict object, mypy checks that each key and value is compatible
+with the TypedDict type that is inferred from the surrounding context.
+
+Example:
+
+.. code-block:: python
+
+    from typing_extensions import TypedDict
+
+    class Point(TypedDict):
+        x: int
+        y: int
+
+    # Error: Incompatible types (expression has type "float",
+    #        TypedDict item "x" has type "int")  [typeddict-item]
+    p: Point = {'x': 1.2, 'y': 4}
+
+
+Check that type of target is known [has-type]
+---------------------------------------------
+
+Mypy sometimes generates an error when it hasn't inferred any type for
+a variable being referenced. This can happen for references to
+variables that have not been initialized yet, and for references
+across modules that form an import cycle. When this happens, the
+reference gets an implicit ``Any`` type.
+
+To work around this error, you can add an explicit type annotation to
+the target variable or attribute. Sometimes you can also reorganize
+the code so that the definition of the variable is placed earlier than
+the reference to the variable in a source file. Untangling cyclic
+imports may also help.
+
+Check that import target can be found [import]
+----------------------------------------------
+
+Mypy generates an error if it can't find the source code or a stub file
+for an imported module.
+
+Example:
+
+.. code-block:: python
+
+    # Error: Cannot find module named 'acme'  [import]
+    import acme
+
+See :ref:`ignore-missing-imports` for how to work around these errors.
+
+Check that each name is defined once [no-redef]
+-----------------------------------------------
+
+Mypy may generate an error if you have multiple definitions for a name
+in the same namespace.  The reason is that this is often an error, as
+the second definition may overwrite the first one. Also, mypy often
+can't be able to determine whether references point to the first or
+the second definition, which would compromise type checking.
+
+If you ignore this error, each reference to the defined name refers to
```

Comment:
ignore -> silence?

Also "each reference ... refers to" -> "all references ... refer to"